viam-cartographer/src/carto_facade/carto_facade.h

```diff
@@ -97,20 +108,23 @@ typedef enum viam_carto_LIDAR_CONFIG {
 #define VIAM_CARTO_DATA_DIR_INVALID_DEPRECATED_STRUCTURE 16
 #define VIAM_CARTO_DATA_DIR_FILE_SYSTEM_ERROR 17
 #define VIAM_CARTO_MAP_CREATION_ERROR 18
-#define VIAM_CARTO_SENSOR_NOT_IN_SENSOR_LIST 19
+#define VIAM_CARTO_UNKNOWN_SENSOR_NAME 19
 #define VIAM_CARTO_LIDAR_READING_EMPTY 20
 #define VIAM_CARTO_LIDAR_READING_INVALID 21
 #define VIAM_CARTO_GET_POSITION_RESPONSE_INVALID 22
 #define VIAM_CARTO_POINTCLOUD_MAP_EMPTY 23
-#define VIAM_CARTO_GET_POINT_CLOUD_MAP_RESPONSE_INVLALID 24
+#define VIAM_CARTO_GET_POINT_CLOUD_MAP_RESPONSE_INVALID 24
 #define VIAM_CARTO_LIB_ALREADY_INITIALIZED 25
-#define VIAM_CARTO_GET_INTERNAL_STATE_RESPONSE_INVLALID 26
+#define VIAM_CARTO_GET_INTERNAL_STATE_RESPONSE_INVALID 26
 #define VIAM_CARTO_GET_INTERNAL_STATE_FILE_WRITE_IO_ERROR 27
 #define VIAM_CARTO_GET_INTERNAL_STATE_FILE_READ_IO_ERROR 28
 #define VIAM_CARTO_NOT_IN_INITIALIZED_STATE 29
 #define VIAM_CARTO_NOT_IN_IO_INITIALIZED_STATE 30
 #define VIAM_CARTO_NOT_IN_STARTED_STATE 31
 #define VIAM_CARTO_NOT_IN_TERMINATABLE_STATE 32
+#define VIAM_CARTO_IMU_ENABLED_INVALID 33
```

Comment:
What does this error mean?

Reply:
This error exists for when the `use_imu_data` has been set to true but no imu has been given, or if `use_imu_data` has been set to false and an imu has been given. As `use_imu_data` is set via the presence or absence of an imu, this should never be hit during normal operation.

Reply:
What do you think about calling it something a little more descriptive of that scenario? I think I would be confused if I got that error. 

Maybe `VIAM_CARTO_IMU_CONFIG_INVALID` or maybe two different errors: `VIAM_CARTO_IMU_ENABLED_BUT_NO_IMU_PROVIDED` & `VIAM_CARTO_IMU_PROVIDED_BUT_IMU_NOT_ENABLED`

Reply:
Since this is an uncommon error I'd prefer to keep it as one error. How about `VIAM_CARTO_IMU_PROVIDE_AND_IMU_ENABLED_MISMATCH`?